src/ui/exile_modal.py

```diff
@@ -0,0 +1,58 @@
+import discord
+from services.exile_service import exile_user
+from util import calculate_time_delta
+from .helper import create_modal_embed
+from commands.helper import create_response_context
+
+
+class ExileModal(discord.ui.Modal):
+    def __init__(self, user: discord.Member) -> None:
+        super().__init__(title=f"Exile User {user.display_name}")
+        self.user = user
+
+    duration = discord.ui.TextInput(label="Exile Duration", required=True)
+
+    reason = discord.ui.TextInput(
+        label="Exile Reason",
+        style=discord.TextStyle.long,
+        placeholder="Reason for exiling user",
+        max_length=512,
+        required=True,
+    )
+
+    async def on_submit(self, interaction: discord.Interaction):
+        exile_duration = calculate_time_delta(self.duration.value)
+
+        if self.duration.value and not exile_duration:
+            # TODO update this string once exile duration change is merged
```

Comment:
let's add the actual ticket number here

Reply:
I'm just gonna hold off on this merge until after the input formatting change goes in. I rewrote this to have the values all over the place and it's not worth the headache of merging this then breaking it again